java/src/client/join/JoinGameController.java

```diff
@@ -117,9 +117,20 @@ public void reloadGamesList() {
                 .onSuccess(games -> SwingUtilities.invokeLater(() -> {
                     getJoinGameView().setGames(games, getGameManager().getPlayerInfo());
                     if (!getNewGameView().isModalShowing() && !getSelectColorView().isModalShowing()) {
-                        //TODO fix this whack redrawing stuff
                         getJoinGameView().closeModal();
                         getJoinGameView().showModal();
+                    } else if (getSelectColorView().isModalShowing() && selectedGame != null) {
+                        for (GameInfo g : games) {
+                            if (g.getId() == selectedGame.getId()) {
+                                selectedGame = g;
+                                break;
+                            }
+                        }
+                        selectedGame.getPlayers().stream()
```

Comment:
Could probably have the above logic integrated into this, but still: _niiiiiice_.